In [1]:
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split
from sklearn.metrics import mean_squared_error
from sklearn.metrics import r2_score
from sklearn.metrics import mean_absolute_error
from sklearn.metrics import mean_absolute_percentage_error

from sklearn import linear_model

In [2]:
# df = pd.read_csv('wc-wo-outliers.csv')
# df2 = pd.read_csv('wc-w-outliers.csv')
df = pd.read_csv('hotData.csv')

In [3]:
y = df['results']
X = df.drop(columns=['results'])

X_train, X_test, y_train, y_test = train_test_split(
     X, y, test_size=0.15, random_state=1)

In [4]:
clf = linear_model.ARDRegression(alpha_1=1000, alpha_2=1000)
# clf2 = linear_model.ARDRegression()

In [5]:
clf.fit(X_train, y_train)
# clf2.fit(X_train2, y_train2)

ARDRegression(alpha_1=1000, alpha_2=1000)

In [6]:
pred_train= clf.predict(X_train)
pred_test= clf.predict(X_test)

In [7]:
# pred_train= clf2.predict(X_train2)
# pred_test= clf2.predict(X_test2)

In [8]:
# print('RMSE : '+str(np.sqrt(mean_squared_error(y_test2,pred_test2)))) 
# print('R2 : '+str(r2_score(y_test2, pred_test2)))
# print('MSE: '+str(mean_squared_error(y_test2, pred_test2)))
# print('MAE: '+str(mean_absolute_error(y_test2, pred_test2)))
# print('MAPE: '+str(mean_absolute_percentage_error(y_test2, pred_test2)))

print('---------------------')

# pred_test= clf.predict(X_test)

print('RMSE : '+str(np.sqrt(mean_squared_error(y_test,pred_test)))) 
print('R2 : '+str(r2_score(y_test, pred_test)))
print('MSE: '+str(mean_squared_error(y_test, pred_test)))
print('MAE: '+str(mean_absolute_error(y_test, pred_test)))
print('MAPE: '+str(mean_absolute_percentage_error(y_test, pred_test)))

---------------------
RMSE : 7.918146501300685
R2 : 0.2893679082847298
MSE: 62.697044016060275
MAE: 5.907462135448373
MAPE: 1.2685683903338687


MAPE: 1.3934502148026755
alpha_1=10 MAPE: 1.3780892330887005
alpha_1=10000 MAPE: 1.2881572628411337
alpha_1=1000 MAPE: 1.270121254581416
alpha_2=1000 MAPE: 1.2685683903338687

In [9]:
data = {'y_test': y_test, 'y_pred': pred_test}
new_df = pd.DataFrame(data)
diff = (new_df['y_test'] - new_df['y_pred']).abs()
new_df['diff'] = diff
new_df = new_df.sort_values(by='diff', ascending=True)


In [10]:
new_df

y_test     y_pred       diff
187      16  15.889359   0.110641
107      27  26.888939   0.111061
89       22  21.637533   0.362467
251      10   9.572754   0.427246
154      22  22.707174   0.707174
169       1   1.793184   0.793184
159       5   5.896314   0.896314
212      13  14.540315   1.540315
274       1   2.638056   1.638056
208      31  29.217436   1.782564
194      21  19.083232   1.916768
259       1   3.176312   2.176312
11       12   9.690459   2.309541
42       17  19.514909   2.514909
266      12   9.384195   2.615805
268       5   7.929957   2.929957
102      13  15.987208   2.987208
192      20  16.853670   3.146330
17       29  25.264203   3.735797
12       27  30.999805   3.999805
179      13  17.033309   4.033309
227      32  27.762847   4.237153
186      12  16.536494   4.536494
88        8  12.983391   4.983391
248       6   1.000137   4.999863
59       13  18.526522   5.526522
171      18  23.901512   5.901512
99        9  15.337082   6.337082
279      19  25.521015   6.521015
90       12  18.916247   6.916247
134      22  15.049382   6.950618
18        2  10.205104   8.205104
158      15   5.797897   9.202103
119       7  16.226176   9.226176
62       32  22.698292   9.301708
217      11  22.050269  11.050269
214      31  18.550799  12.449201
132      28  15.430061  12.569939
27        4  16.597500  12.597500
258       9  -4.011835  13.011835
105       1  18.536246  17.536246
138      19   0.424156  18.575844
4         2  24.650286  22.650286

In [11]:
new_df['diff'].mean()

5.907462135448374

In [12]:
new_df['diff'].median()

4.23715315344845